finality-provider/service/app.go

```diff
@@ -237,73 +240,65 @@ func (app *FinalityProviderApp) getFpPrivKey(fpPk []byte) (*btcec.PrivateKey, er
 	return record.PrivKey, nil
 }
 
-// SyncFinalityProviderStatus syncs the status of the finality-providers
-func (app *FinalityProviderApp) SyncFinalityProviderStatus() error {
-	var (
-		latestBlock *types.BlockInfo
-		err         error
-	)
-
-	attempts := uint(app.config.MinutesToWaitForConsumer)
-	err = retry.Do(func() error {
-		latestBlock, err = app.cc.QueryBestBlock()
-		if err != nil {
-			return err
-		}
-		return nil
-	}, retry.OnRetry(func(n uint, err error) {
-		app.logger.Debug(
-			"failed to query the consumer chain for the latest block",
-			zap.Uint("attempt", n+1),
-			zap.Uint("max_attempts", attempts),
-			zap.Error(err),
-		)
-		// waits for consumer chain to become online for one week
-		// usefull to turn fpd on before the consumer chain node is available
-		// and waiting for the consumer node to become available to start.
-	}), retry.Attempts(attempts), retry.Delay(time.Minute), RtyErr)
+// SyncFinalityProviderStatus syncs the status of the finality-providers with the chain.
+func (app *FinalityProviderApp) SyncFinalityProviderStatus() (fpInstanceStarted bool, err error) {
+	latestBlock, err := app.cc.QueryBestBlock()
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	fps, err := app.fps.GetAllStoredFinalityProviders()
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	for _, fp := range fps {
 		vp, err := app.cc.QueryFinalityProviderVotingPower(fp.BtcPk, latestBlock.Height)
 		if err != nil {
-			// if error occured then the finality-provider is not registered in the Babylon chain yet
+			// if error occured then the finality-provider is not registered in the Babylon chain yet or
+			// there is nothing in the voting power table, so it should not start the fp.
+			allowedErr := fmt.Sprintf("failed to query Finality Voting Power at Height %d: rpc error: code = Unknown desc = %s: unknown request", latestBlock.Height, bstypes.ErrVotingPowerTableNotUpdated.Wrapf("height: %d", latestBlock.Height).Error())
+			if !strings.EqualFold(err.Error(), allowedErr) {
```

Comment:
```suggestion
			if !errors.Is(err, bstypes.ErrVotingPowerTableNotUpdated) {
```

Reply:
I was not able to do it with `error.Is` due to wrapping of rpc error of cosmos RPC handler